### PR TITLE
Issue #196 : Allow to cancel an upload

### PR DESCRIPTION
**Pull request by DimitriGilbert**
The button isn't fancy but i did not find directive concerning visual elements. Tell me what to change it for, i'll be happy to do so.

### CODE REVIEW

File: client/src/app/videos/+video-edit/video-add.component.ts [hunk 1]
```diff
@@ -27,6 +27,7 @@ export class VideoAddComponent extends FormReactive implements OnInit {
 
   isUploadingVideo = false
   videoUploaded = false
+  videoUpload = null
```
Comment (bnjbvr) [inline client/src/app/videos/+video-edit/video-add.component.ts]: From a quick skim, isn't `videoUpload` redundant with `isUploadingVideo`, and thus we could remove the latter by testing `!!videoUpload`

Comment (DimitriGilbert) [reply client/src/app/videos/+video-edit/video-add.component.ts]: As I do not know the code base very well yet, I didn't want to introduce unforeseen side-effect. Also, isVideoUploading is a boolean flag while videoUpload (very bad name but I was out of idea) is the observable returned by the subscription to the HttpClient request and I was not feeling comfortable testing on it (the upload is finished but for some reason, something did not go entirely as planned for example).
I'm not against it though. 
What do maintainers thinks about that ?

Comment (Chocobozzz) [reply client/src/app/videos/+video-edit/video-add.component.ts]: Just rename it to `videoUploadObservable` and it should be okay :)

File: client/src/app/videos/+video-edit/video-add.component.ts [hunk 2]
```diff
@@ -93,6 +94,16 @@ export class VideoAddComponent extends FormReactive implements OnInit {
     return this.form.valid
   }
 
+  uploadCancel () {
```
Comment (Chocobozzz) [inline client/src/app/videos/+video-edit/video-add.component.ts]: `cancelUpload`?

**Comment by DimitriGilbert** [comment]
I did do the requested change, I also added a few paths to the gitignore, they seem to be created on the fly and contain user related content. Would you like me to open another pull request for that  and delete the changes on this one ?

**Comment by Chocobozzz** [comment]
@DimitriGilbert No it's fine. The directories are automatically created according to your configuration: https://github.com/Chocobozzz/PeerTube/blob/develop/config/default.yaml#L19 

Thanks for your work! :)

**Comment by DimitriGilbert** [comment]
You're welcome :)

**Pull request merged**